### ISSUE TITLE
iocpreactor.tcp.ClientSocket should implement ISystemHandle

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2021]
|[<img alt="ghazel's avatar" src="https://avatars.githubusercontent.com/u/31402?s=50" width="50" height="50">](https://github.com/ghazel)| @ghazel reported|
|-|-|
|Trac ID|trac#2021|
|Type|enhancement|
|Created|2006-08-28 08:51:31Z|

Dead simple, branching to implement it now

<details><summary>Searchable metadata</summary>

```
trac-id__2021 2021
type__enhancement enhancement
reporter__ghazel ghazel
priority__highest highest
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__duplicate duplicate
component__core core
keywords__windows__iocpreactor windows, iocpreactor
time__1156755091000000 1156755091000000
changetime__1205873666000000 1205873666000000
version__None None
owner__ghazel ghazel

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ghazel's avatar" src="https://avatars.githubusercontent.com/u/31402?s=50" width="50" height="50">](https://github.com/ghazel)<a name="note_1"></a>|@ghazel commented|
|-|-|

iocp_syshandle-2021

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun set owner to @ghazel|
|-|-|

implements() can only be used once in a class definition, so the udp.py changes are broken.

There should be some tests for this code.  Set up a connection and call the method to make sure it doesn't raise an exception, perhaps?  Not a very good test, but better than nothing.

**Comment by twisted-trac** [comment]
|<img alt="PenguinOfDoom's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_4"></a>|PenguinOfDoom set status to `closed`|
|-|-|

Implemented in [#1760](https://github.com/twisted/twisted/issues/1760)
